### PR TITLE
feat(tools): export Alertprops if needed for consumption

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

There is one Alert that has `Close button` in it, in the codebase, so I am happy to close this and leave a note for it.

https://github.com/freeCodeCamp/freeCodeCamp/blob/2d41346d7956006a732580166771c8490c09c183/client/src/components/Flash/index.tsx

```ts
    <Alert
          bsStyle={type}
          className='flash-message'
          closeLabel={t('buttons.close')}
          onDismiss={handleClose}
        >
          {t(message, variables)}
        </Alert>
  ```

### CODE REVIEW

File: tools/ui-components/src/alert/alert.tsx [hunk 1]
```diff
@@ -35,7 +40,12 @@ export function Alert({
 
   return (
     <div className={classes} role='alert'>
+      {canDismiss ? (
+        <CloseButton onClick={onDismiss} label={closeLabel ?? 'Close Alert'} />
```
Comment (naomi-lgbt) [inline tools/ui-components/src/alert/alert.tsx:44]: Should we make the default fallback translatable?

**Review by naomi-lgbt — DISMISSED** [review]

File: tools/ui-components/src/alert/alert.tsx (hunk 1 above)
Comment (Sboonny) [reply tools/ui-components/src/alert/alert.tsx:44]: is `useTranslation` can be imported? 🤔

Let me check in the morning, if not we may need to discuss this with Ahmad.

Comment (Sboonny) [reply tools/ui-components/src/alert/alert.tsx]: I don't think it's possible to use `useTranslate`.

For now, I have removed the value, and made it if someone use `onDismiss`, they have to provide a value, so it isn't possible to have `onDismiss`, without `closeLabel` and vise-versa.

This doesn't solve the issue of untranslatable strings in the ui-component, but this forces us to provided a string (translatable or not) in the `/learn`, when using the component.

We will still have to discuss this with Ahmad, on how he want this untranslatable strings to be tackled.

**Comment by ahmadabdolsaheb** [comment]
Here is a pr that made Alert not dismissible: https://github.com/freeCodeCamp/freeCodeCamp/pull/45291
Please look into the link discussion and see if it is still relevant. 
If not, we can bring back the tests that have been removed in the above-mentioned pr.

**Comment by Sboonny** [comment]
Nope, just thought that we should delay remaking the flash and decoupling the alert after the library is published, but this isn't a big change that we should care about it.

We can easily add closebutton to flash component when the library is published, and it won't have issues. https://github.com/freeCodeCamp/freeCodeCamp/blob/2d41346d7956006a732580166771c8490c09c183/client/src/components/Flash/index.tsx

File: tools/ui-components/src/alert/alert.tsx (hunk 1 above)
Comment (Sboonny) [reply tools/ui-components/src/alert/alert.tsx]: I will mention this in today meeting, and open issue about what we come up with.

**Comment by Sboonny** [comment]
Removed the close button component and exported the type, in the chance we need to use it, when we creating different component in the codebase

File: client/src/components/SolutionViewer/SolutionViewer.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ function SolutionViewer({ challengeFiles, solution }: Props): JSX.Element {
       {solutions.map(({ fileKey, ext, contents }) => (
         <Panel bsStyle='primary' className='solution-viewer' key={fileKey}>
           <Panel.Heading>{ext.toUpperCase()}</Panel.Heading>
+          <p>here is a test string</p>
```
Comment (Sboonny) [inline client/src/components/SolutionViewer/SolutionViewer.tsx:30]:
```suggestion
```

File: tools/ui-components/src/alert/alert.tsx [hunk 3]
```diff
@@ -20,13 +20,12 @@ const variantClasses: Record<AlertVariant, string> = {
...
  * `Alert` is not dismissable.
  */
-export function Alert({
+export const Alert = ({
   children,
   className,
   variant
-}: AlertProps): JSX.Element {
+}: AlertProps): JSX.Element => {
   const variantClass = variantClasses[variant];
```
Comment (Sboonny) [inline tools/ui-components/src/alert/alert.tsx:28]:
```suggestion
  const variantClass = variantClasses[variant];
  
```

File: tools/ui-components/src/alert/alert.tsx [hunk 4]
```diff
@@ -20,13 +20,13 @@ const variantClasses: Record<AlertVariant, string> = {
...
-
+
```
Comment (Sboonny) [inline tools/ui-components/src/alert/alert.tsx:29]:
```suggestion

```

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Review by ahmaxed — APPROVED** [review]

File: tools/ui-components/src/alert/index.ts [hunk 5]
```diff
@@ -1 +1,2 @@
 export { Alert } from './alert';
+export type { AlertProps } from './alert';
```
Comment (ShaunSHamilton) [inline tools/ui-components/src/alert/index.ts:2]: For future reference, we have an infrequent job that clears all unused exports. So, unless something is "consumed", it should not be "just made to be exported".

I am not sure if this is already imported elsewhere.

Comment (Sboonny) [reply tools/ui-components/src/alert/index.ts:2]: I can see the ui-component become unusable because of that, will this apply to ui-component

**Pull request merged**